mod_project/project.py

```
@@ -471,49 +468,150 @@ def add_layer(fname, target='game-graph'):
+            raise ValueError(f"Mod ID '{mod_id}' is reserved")
+        if not mod_name:
+            raise ValueError("Mod display name cannot be empty")
+        if complete_map and loader_kind != 'fuse':
+            raise ValueError("Complete standalone maps require loader='fuse'")
+        if complete_map:
+            map_origin_lat = float(map_origin_lat)
+            map_origin_lon = float(map_origin_lon)
+            map_tile_dimension = float(map_tile_dimension)
+            if not all(math.isfinite(value) for value in (
+                    map_origin_lat, map_origin_lon, map_tile_dimension)):
+                raise ValueError("Map origin and tile dimension must be finite")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)